### DISCUSSION TITLE
Welcome to discussions!

### DISCUSSION THREAD

**Discussion by zuckung** [zuckung/endless-sky-plugins#16] [Announcements]
Leave a comment. Tell me what you like in my plugins, or hate. What you want to get added, or removed.
Every feedback is welcome!

**Comment by MeretiJane** [comment]
For the most part I like your plugins, but the AI images you use for some of them are well, bad.

**Comment by zuckung** [comment]
first, thx for the feedback!

all of my images are from ai, about which images are you talking? and in which way bad?

**Reply by MeretiJane** [reply]
Well scene images tend to have many defects and sprites are just completely incomprehensible.
For example 1 is a scene image from gateways, during the news of the gateway construction. Many parts of the image are simply uncanny and others are actual defects. The badger sprite in image 2 for example is just incomprehensible nonsense in the general shape of a starship. All other AI images tend to be like this.
The problem with the images generated by AIs is that the AI doesn't understand what it is making. It compiles similarly tagged images and generates something that looks vaguely like the input. The AI has no comprehension of what a starship would be, it doesn't understand the concepts or the parts, so it just uses human made images as reference and blends them. This lack of understanding from the AI causes the images produced to be completely nonsense.
So basically the AI is stupid and doesn't even know what it is trying to make, causing the images to look anywhere from bad to complete nonsense.
![1](https://github.com/user-attachments/assets/593d6840-3f23-4015-b588-fe081b06bad1)
![2](https://github.com/user-attachments/assets/66c8c875-b4f0-4e3c-820d-84021d9802f0)

**Reply by zuckung** [reply]
i see. i know of some of these problems, probably the much more obviously ones.
i.e. i didn't see anything in this special scene image till now. actually i still think 3 of your 5 markings aren't really visually wrong.

but anyway, i have no other choice to taking ai images. it's nearly impossible to find useable real images that can fit in sci fi scenes, and even if i find a fitting image, the copyright is too restrictive. the time for searching things that might optically just fit in a little bit is massive. you can see that in original ES images(scenes/landing images), they are so boring and totally don't fit in a sci fi game(except for a few of the nasa ones).
another point is that i m making plugins on a tablet computer with very limited ressources. even if i were a genius with graphics, and i m totally shitty with that, i had problems making images that fit in to the sci fi flavour, i imagine.

so i m sorry to inform you that i can't fix this problem, nor do i see a way that might be possible, unless a brilliant professional computer graphics artist(the kind who gets paid for creating game images) shows up and delivers me new images.^^ do you see a way?

i hope these graphical issues ruin just a little part of the fun you have with my plugins. :)

**Reply by VaelophisNyx** [reply]
you DO have the choice of just...not using them, you know
its easier and immediately would improve the quality of all your mods

**Reply by zuckung** [reply]
imo without pics the ES experience is worse. and according to the amount of my downloads and a few people i talked about that, the images spice the plugins up.

well, it's everyones own choice to use/like or just don't use/like how ai is used in these plugins.

**Reply by MeretiJane** [reply]
Personally I don't like them, so I just remove the AI images manually.

Anyways, that's not what I'm here for now. I think you should add korath ammo to the Efreti shipyard plugin so that their secondary weapons can actually be viable to use. Would be cool and a simple update.

**Reply by MeretiJane** [reply]
Also I think being able to straight up buy an echo galleon is a bit odd, maybe it would be cool if you had to unlock a license for it or something, so like you could buy their other tech and ships, but they arent just gonna sell a world ship to a total stranger. Like idk some mission chain you do for them.

**Reply by MeretiJane** [reply]
I updated the file if you want to use it
[shipyard.txt](https://github.com/user-attachments/files/17612331/shipyard.txt)

**Reply by MeretiJane** [reply]
oh also they should sell the digger turret too

**Reply by zuckung** [reply]
sure, np, i will add ammo.

actually there are only weapons for sale that the efreti use on their ships. the mining turret is an exiles weapon, clearly stating in its decription.

and regarding the missions ... i thought about that when i created that plugin, but there is the problem with the speaking. and later with rek translating, which have to be an option in the plugin, but would be too late for the normal user. maybe a quarg can translate, or you learn some sign language. well, it's complicated.
writing texts is the part i hate when making plugins, but hey i can use chatGPT :D
fun asides, if you want to write something like this, feel free, you would get full credits.

**Comment by andtim1** [comment]
Hey, i'm trying to learn blender, and since major issues/bugs been fixed, ill try to make some outfits in blender (to replace AI) :)

happy holidays

**Reply by andtim1** [reply]
as it turns out im using a potato for a pc blender was NOT having it :(

_well ill just try to make better ai prompts_

**Comment by ken1442** [comment]
Hi! I really like your mods. You've added a ton of great content to a gem of a game. Your ship merging mod is really unique. Do you have any plans to incorporate merging functionality to other plugins that add ship content, like Mega Freight? I've attempted to edit plugin files manually to merge one ship from the MF mod, and it looks like I'm not quite hitting the mark yet. I'll keep at it. Thanks for supporting the ES community!

[https://github.com/1010todd/Mega-Freight]

**Reply by TheGiraffe3** [reply]
There's a script to generate a ship.merging plugin from a data folder here: https://github.com/zuckung/endless-sky-plugins/blob/main/tools/ES_plugin_script_ship_merging/ES_plugin_script_ship_merging.py
And if you'd prefer not to use that script for whatever reason, I have a GitHub-based version here (instructions in the README): https://github.com/TheGiraffe3/zuckung-scripts-online
There appears to be a compatibility plugin already generated on the [ship.merging releases page](https://github.com/TheGiraffe3/zuckung-scripts-online/releases/tag/ship.merging) of my repo. Note that you'll need to be using the newest version of MGF for it.

**Reply by zuckung** [reply]
Thanks for your feedback, always nice to hear :)

The giraffe was faster :)
and yes, the wish for such a plugin-compatibility already came up several times, for ship.merging and for mission.helper. luckily these plugins are script generated and a small adjustment to the script can make this possible. with the help of the giraffe the scripts are very easy to use, without local coding environment. have fun with it ;)

**Reply by ken1442** [reply]
Wow, that was quick! I really appreciate the help. Keep being awesome!

**Comment by JonasTam** [comment]
Hey, nice plugins, I've install several of them to try them out.

I'm tryin to merge ships, but I cannot find any button or mission or anything like that for the merge. How should I find the merge menu?

**Reply by TheGiraffe3** [reply]
There's a new system called Dent, linked to Cardax. Go there and land on Trillinox, then visit the job board to accept the mission.

**Reply by JonasTam** [reply]
Yep, I'm on this Ship, and it does not give me any options of merging. I have 40 same ships to merge.
<img width="412" height="255" alt="image" src="https://github.com/user-attachments/assets/cc92c0d8-69eb-4b4f-a720-5f2599592606" />

**Comment by zuckung** [comment]
you can find the merging on the job board. but this plugin needs the same ship data as your ES version, else the job doesnt show up.

ship.merging 1.0.7 with ES 0.11.1
ship.merging 1.0.6 with ES 0.11.0
ship.merging 1.0.5 with ES 0.10.16

**Reply by JonasTam** [reply]
Thanks!
How/where i can get the 1.0.5 version? (GIT is still new to me)

**Reply by zuckung** [reply]
<img width="1149" height="589" alt="Screenshot 2026-06-21 175621" src="https://github.com/user-attachments/assets/96ccd7bc-6fa2-46b4-bb51-780d2c33d96c" />

and the info how to start it, within the Dent system, you can see when clicking on the "show readme".

**Reply by JonasTam** [reply]
Oh, thanks! Already managed through GIT (was something new to learn!)

**Comment by rogue-ronin** [comment]
I'm trying your additional.command.buttons.radial. Doesn't offer buttons on linux, but the flagship graphic and putting the fleet ships along the bottom is very nice!

Some small bugs in the planet layout... Could you make a simpler one that does the flagship and fleet stuff without touching the planet part or trying to add the buttons? Love this game, thanks for the cool plugins!

**Reply by zuckung** [reply]
this plugin is made for android version (only), and is not working correctly with windows/linux/mac. i would have already ported it to the PC version, but the hardcoded button handling is different. there is currently no chance getting it to work on PC.

regarding the flagship hud i can help you with https://github.com/tobersj/Central-HUD/tree/main , from there i got it.

i hope that helps at least a bit ;)

**Reply by rogue-ronin** [reply]
I did remove it yesterday after trying (and failing) to move things via the planet.txt file. (I had indeed noted that your readme explained it was for Android only.)

The Central HUD looks great and I will install it shortly. I like the relocation of the "radar" in the screenshot...You have a note in the readme that sometimes the long messages get obscured - could you move the ship to the lower right instead of the middle? I generally use 16:10 monitors and locate my control surfaces and data to the sides. Seems natural somehow - that's where my hands are, yeah? :D

Much aloha and thank you for adding to the game!

**Reply by rogue-ronin** [reply]
Whoops! just noticed that is not your repository, and it is not in the eslauncher plugins list. I'll learn how to install it manually and please ignore the request to mess with the flagship location! :laughing:
